### ISSUE TITLE
Windows version does not deploy to cloudsmith after PR #50

### ISSUE THREAD

**Issue by mauroc** [mauroc/squiddio_pi#61]
things work great after merging @jongough's PR#50, except that the Windows version does not deploy to Cloudsmith : Here's the relevant message in the appveyor log

```
bash ../ci/appveyor-upload.sh
fatal: ref HEAD is not a symbolic ref
Not on master branch, skipping deployment.
```
This build/deploy was triggered by a push on the master branch, so not sure what's going on. ...

**Comment by rgleason** [comment]
The message on the failure is different now, 
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```
Possibly the build was terminated due to more recent builds starting?  
Perhaps a setting in circleci that saves on unnecessary labor.

**Comment by mauroc** [comment]
the issue I reported affects Appveyor -> Cloudsmith
Your message appears to relate to Travis. I have noticed the OSX build on Travis stalls at times, and/or can take more than 10 minutes. 
Not sure about the comment on circleci. My circleci builds all deploy successfully.

**Comment by rgleason** [comment]
Sorry, missed that.

**Comment by leamas** [comment]
This is most likely caused by appveyor for some reason  checking out a detached head branch. The culprit is the `branch=$(git symbolic-ref --short HEAD` line in appveyor-upload.sh, part of the new test for being on master branch or not.

I'll try to take a look later.

**Comment by leamas** [comment]
Here is an *untested* patch if anyone cares to test it...


[0001-ci-appveyor-upload-Fix-missing-git-HEAD-61.patch.gz](https://github.com/mauroc/squiddio_pi/files/3746857/0001-ci-appveyor-upload-Fix-missing-git-HEAD-61.patch.gz)

In haste...

**Comment by rgleason** [comment]
I downloaded it. Now I expand it into my current local dir reset35-rg? Yes I think so.

**Comment by rgleason** [comment]
from (current mauroc) master checked out a new branch leamas-patch-win-deploy
Moved the patch into a subdirectory "patch" and expanded the gz
```
C:\Users\Rick\Documents\GitHub\squiddio_pi>git am patches/0001-ci-appveyor-upload-Fix-missing-git-HEAD-61.patch
Applying: ci: appveyor-upload: Fix missing git HEAD (#61)
```
It builds locally.  Considering the issues to  git push -u origin leamas-patch-deploy-win
1. This might deploy all the way up to circleci.
2. What will happen when I delete this branch and remove it from my public repository?

**Comment by rgleason** [comment]
Ok, I guess that patch worked!  Thank you Alec. 
https://github.com/rgleason/squiddio_pi/runs/266850915
using a new branch https://github.com/rgleason/squiddio_pi/commits/leamas-patch-win-deploy

Now what will happen when I remove/delete the branch locally and in rgleason/squiddio_pi?  Something's going to happen in circleci I think.

**Comment by rgleason** [comment]
Well, actually it did not deploy because it was not on the master branch.
```
#!/bin/bash -eo pipefail
ci/circleci-upload.sh
Not on master branch, skipping deployment.
```
https://circleci.com/gh/rgleason/squiddio_pi/17?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

but it looks to me like it will work on mauro's master

**Comment by rgleason** [comment]
I don't want to try this on my master branch because, I would like to keep it identical  to mauro's master.
Also if I commit something and it gets run by circleci and uploaded to cloudsmith, and then I revert that commit, what happens inthe cloudsmith and circleci services?

**Comment by leamas** [comment]
> I don't want to try this on my master branch because, I would like to keep it identical to mauro's master.

You should make a pull request. See for example https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

>  what happens in cloudsmith 

Nothing, you can't deploy anything since only ·@mauroc has the deployment key

> and circleci services?

ci servers  don't care about the history. They just build whatever you push.

**Comment by rgleason** [comment]
> You should make a pull request. See for example https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

Nice resource, wish I had that when I got started years ago.

> Nothing, you can't deploy anything since only ·@mauroc has the deployment key

Really?  So for all the orphaned, intermittent and occasional Plugin Authors, we are going to rely on them to deploy?  We would be stuck back at v5.0 without many plugins if that is the case. I have made this clear before.

I assume you are saying I can't deploy to his **Cloudsmith directory**,  because of the way it's been set up.

> ci servers don't care about the history. They just build whatever you push.

Yes, I guess that is true, because Cloudsmith and CircleCI are not keeping track of "Commits" like Github does.  This is both an asset and a liability. What it means is I can delete/remove a test branch that has been deployed to my Cloudsmith account through CircleCI and not have any bad things happen, provided nobody else is reliant on it.
